examples/empirical_frechet_mean_uncertainty_s3.py

```diff
@@ -0,0 +1,184 @@
+"""
+Uncertainty of the empirical Fréchet mean on the sphere
+
+The variance of the Fréchet mean FM_n of a sample of n IID
+random variables of variance Var is decreasing more slowly
+in a sphere than in a Euclidean space.
+This example computes the  modulation factor
+     alpha = Var( FM_n) / ( n * Var)
+for isotropic distributions on hyperspheres or radius 0<r<Pi
+in the sphere s3.
+"""
+
+
+import matplotlib.pyplot as plt
+import numpy as np
+
+import geomstats.backend as gs
+# import geomstats.visualization as visualization
+
+from geomstats.geometry.hypersphere import Hypersphere
+
+# n = number of samples for computing the mean
+# NN = number of trials for the stochastic integral of expectation
+# Ntheta = number of points on the curve sig_est(theta)
+
+def my_random_uniform_sphere(n):
+    #locals(x,xn)
+    # Random coordinates between -1 and 1
+    x = (np.random.random_sample(n+1)*2.0 -1.0)
+    nx = np.linalg.norm(x)
+    while nx > 1 or nx < 1.e-10:
+       x = (np.random.random_sample(n+1)*2.0 -1.0)
+       nx = np.linalg.norm(x)
+    return(x/nx)
+
+def my_random_sample(N, dim):
+    X = np.zeros( (N, dim+1), dtype=gs.float64)
+    for i in range(N):
+        X[i,:] = my_random_uniform_sphere(dim)
+    return X
+
+
+
+def empirical_var(n, theta, dim, NN=1000):
+    """
+    Compute the variance of the empirical Fréchet mean of n samples
+    drawn from an isotropic distributions on hyper sub-spheres of radius
+    0 < theta < Pi around the north pole of the sphere of dimension dim.
+    This is repeated NN times to make a stochastic approximation of the
+    expectation.
+    """
+    assert dim > 1, "Dim > 1 needed to draw a uniform sample on subsphere"
+    var = []
+    sphere = Hypersphere(dimension=dim)
+    subsphere = Hypersphere(dimension=dim-1)
+
+    # Define north pole
+    north_pole = np.zeros(dim+1)
+    north_pole[dim] = 1.0
+    for j in range(NN):
+        # Sample n points from the uniform distrib on a subsphere
+        # of radius theta (i.e cos(theta) in ambiant space)
+        data = gs.zeros((n, dim+1), dtype=gs.float64)
+        # For sampling on a subsphere, use RandomUniform(dim-1)
+        directions = subsphere.random_uniform(n)
+        # directions = my_random_sample(n, dim-1)
```

Comment:
I personally like to keep alternative options commented in my code, but I understand that it may be difficult to read in general. This has been removed in this example. However, I believe that this may push people (including me) to keep they half-cooked code apart because only the fully cooked one can be included in the library.